paddle/fluid/pir/transforms/onednn/matmul_activation_fuse_pass.cc

```diff
@@ -0,0 +1,704 @@
+
+    const auto &full1 =
+        pat.Op(paddle::dialect::FullOp::name(),
+               {{"shape", pat.Attr("shape1")}, {"value", pat.Attr("value1")}});
+    const auto &full2 =
+        pat.Op(paddle::dialect::FullOp::name(),
+               {{"shape", pat.Attr("shape2")}, {"value", pat.Attr("value2")}});
+    pat.Tensor("min") = full1();
+    pat.Tensor("max") = full2();
+
+    const auto &act = pat.Op(paddle::dialect::ClipOp::name());
+    matmul({&pat.Tensor("X"), &pat.Tensor("Y"), &pat.Tensor("residual")},
+           {&pat.Tensor("Out")});
+
+    pat.Tensor("act_out") =
+        act(pat.Tensor("Out"), pat.Tensor("min"), pat.Tensor("max"));
+
+    pat.RequireNativeCall([&](const paddle::drr::MatchContext &match_ctx) {
+      std::set<bool> bool_sets = {true, false};
+      auto result_x = match_ctx.Attr<bool>("transpose_x");
+      auto result_y = match_ctx.Attr<bool>("transpose_y");
+      auto act_type = match_ctx.Attr<std::string>("fuse_activation");
+      if (bool_sets.count(result_x) == 0 || bool_sets.count(result_y) == 0 ||
+          act_type != "") {
+        return false;
+      }
+      return true;
+    });
+
+    paddle::drr::ResultPattern res = pat.ResultPattern();
+
+    std::unordered_map<std::string, paddle::drr::Attribute> fused_attrs{
+        {"trans_x", pat.Attr("transpose_x")},
+        {"trans_y", pat.Attr("transpose_y")},
+        {"matmul_alpha", pat.Attr("matmul_alpha")},
+        {"fuse_activation", res.StrAttr("clip")},
+        {"fuse_alpha", pat.Attr("value1")},
+        {"fuse_beta", pat.Attr("value2")},
+        {"fused_output_scale", pat.Attr("fused_output_scale")},
+        {"fused_reshape_x", pat.Attr("fused_reshape_x")},
+        {"fused_transpose_x", pat.Attr("fused_transpose_x")},
+        {"fused_reshape_y", pat.Attr("fused_reshape_y")},
+        {"fused_transpose_y", pat.Attr("fused_transpose_y")},
+        {"fused_reshape_out", pat.Attr("fused_reshape_out")},
+        {"fused_transpose_out", pat.Attr("fused_transpose_out")},
+        {"mkldnn_data_type", pat.Attr("mkldnn_data_type")},
+        {"scale_x", pat.Attr("scale_x")},
+        {"scale_y", pat.Attr("scale_y")},
+        {"scale_in_eltwise", pat.Attr("scale_in_eltwise")},
+        {"scale_out", pat.Attr("scale_out")},
+        {"force_fp32_output", pat.Attr("force_fp32_output")}};
+
+    const auto &fused_matmul = res.Op(fused_matmul_name_, fused_attrs);
+
+    fused_matmul({&res.Tensor("X"), &res.Tensor("Y"), &res.Tensor("residual")},
+                 {&res.Tensor("act_out")});
+  }
+};
+
+class MatmulActivationFusePass : public pir::PatternRewritePass {
+ public:
+  MatmulActivationFusePass()
+      : pir::PatternRewritePass("matmul_activation_fuse_pass", 3) {}
+
+  pir::RewritePatternSet InitializePatterns(pir::IrContext *context) override {
+    pir::RewritePatternSet ps(context);
+    // std::vector<bool> bool_set = {false, true};
+    int benefit_idx = 1;
+    for (auto act_op : act_ops) {
```

Comment:
```suggestion
    for (const auto& act_op : act_ops) {
```
另外关于benefit_idx的设置可能有些问题，从代码逻辑上看，从上到下依次添加的这些pattern，他们的benefit_idx是依次变大的，benefit_idx值越大的pattern，我们期望是越优先被应用，以MatmulClipFusePattern和FusedMatmulClipFusePattern来说，我们期望MatmulClipFusePattern比FusedMatmulClipFusePattern先应用，因此MatmulClipFusePattern的benefit_idx应该较大才对

Reply:
从我的理解来看，因为这个pass目前matmul相关pass里不属于第一个pass，也就是说，在之后遇到fused_matmul的可能性较大，所以这种情况下优先match fused_matmul会不会更好一点？

Reply:
不过如果是以单个pattern为准在整个graph上取多个符合要求的pattern，那确实应该是原生matmul+act被match到的个数更多的可能性较大

Reply:
目前这样也可以，问题不大

Reply:
好的，那我就先不改了哈。如果之后pass order有变化或者说还是需要以非fused_matmul的pattern为主我再随时更改~